fix: Windows compatibility and encoding issues for non-English locales

IDontHaveBrain opened the pull request:
This PR addresses several issues that prevented proper installation and operation on Windows systems, particularly for users with non-English locales:

Fixed path handling in regex patterns to support both forward and backslash path separators
Added Windows-specific command execution for tree-sitter CLI through cmd.exe
Implemented proper handling of include paths in C source files on Windows
Most importantly, added critical MSVC compiler options to resolve encoding issue:

/utf-8 to force UTF-8 encoding for source files regardless of system locale

These changes ensure the package can be successfully built and installed on Windows systems using non-UTF8 code pages (CP949, CP932, etc.).

Fixes #14

Goldziher commented on scripts/clone_vendors.py:
```diff
@@ -79,9 +82,18 @@ async def handle_generate(language_name: str, directory: str | None, abi_version
         else (vendor_directory / language_name).resolve()
     )
+    
+    # 윈도우 환경에서는 cmd.exe /c를 통해 명령어를 실행해야 PATH에서 tree-sitter를 찾을 수 있음
```
please use english or remove the comment - also it should only be essential, if the comment can be understood from the code, no comment is required.

IDontHaveBrain commented:
I've addressed all the feedback in this PR. Windows testing has been added to the GitHub Actions workflow, and now CI runs on both Ubuntu and Windows environments to ensure cross-platform compatibility. I've also fixed the validation errors by replacing the Python 3.12-specific relative_to method with   
walk_up parameter with the more compatible os.path.relpath() function.

Thank you for maintaining this excellent library! We're actively using tree-sitter-language-pack in our projects and truly appreciate the comprehensive language support it provides.

Goldziher commented:
> I've addressed all the feedback in this PR. Windows testing has been added to the GitHub Actions workflow, and now CI runs on both Ubuntu and Windows environments to ensure cross-platform compatibility. I've also fixed the validation errors by replacing the Python 3.12-specific relative_to method with   
> walk_up parameter with the more compatible os.path.relpath() function.
> 
> Thank you for maintaining this excellent library! We're actively using tree-sitter-language-pack in our projects and truly appreciate the comprehensive language support it provides.

Perfect, thanks 🙏
